nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/main/java/org/nuxeo/ecm/platform/comment/impl/AbstractCommentManager.java

```diff
@@ -193,4 +205,80 @@ protected void notifyEvent(CoreSession session, String eventType, DocumentModel
     }
 
+    /**
+     * Checks if a document has comments.
+     *
+     * @param session the core session
+     * @param document the document model who's comments are being counted
+     * @return true if comments were found
+     * @since 11.1
+     */
+    protected boolean hasComments(CoreSession session, DocumentModel document) {
+        // CommentManagerImpl doesn't support it.
+        throw new UnsupportedOperationException();
+    }
+
+    /**
+     * Checks if a document has comments from a particular user.
+     *
+     * @param session the core session
+     * @param document the document model who's comments are being counted
+     * @return true if comments were found
+     * @since 11.1
+     */
+    protected boolean hasCommentsByUser(CoreSession session, DocumentModel document, String author) {
+        // CommentManagerImpl doesn't support it.
+        throw new UnsupportedOperationException();
+    }
+
+    /**
+     * Subscribes top level document's author on first comment and commentor on his first comment if autosubscribtion is
+     * turned on.
+     *
+     * @param session the core session
+     * @param commentDocModel the comment
+     * @param document the document being commented
+     * @since 11.1
+     */
+    protected void handleNotificationAutoSubscriptions(CoreSession session, DocumentModel commentDocModel,
+                                                       DocumentModel document) {
+        if (Framework.getService(ConfigurationService.class).isBooleanFalse(AUTOSUBSCRIBE_CONFIG_KEY)) {
+            return;
+        }
+
+        NuxeoPrincipal topLevelDocumentAuthor = getAuthor(document);
+        if (!hasComments(session, document)) {
+            // Document author is subscribed on first comment by anybody
+            subscribeCommentAddedNotification(document, topLevelDocumentAuthor);
+
+        }
+
+        NuxeoPrincipal commentAuthor = getAuthor(commentDocModel);
+        if (topLevelDocumentAuthor.getName().equals(commentAuthor.getName())) {
+            // Document author is comment author. He doesn't need to be resubscribed
+            return;
+        }
+
+        if (!hasCommentsByUser(session, document, commentAuthor.getName())) {
+            // Comment author is writing his first comment on the document
+            subscribeCommentAddedNotification(document, commentAuthor);
+        }
+    }
+
+    /**
+     * Subscribes a user to "CommentAdded" notifications on the document.
+     *
+     * @param document the document being commented
+     * @param user the user to subscribe to comment notifications
+     * @since 11.1
+     */
+    protected void subscribeCommentAddedNotification(DocumentModel document, NuxeoPrincipal user) {
+        String subscriber = NotificationConstants.USER_PREFIX + user.getName();
+        NotificationManager notificationManager = Framework.getService(NotificationManager.class);
+        if (notificationManager != null) {
```

Comment:
is there any reason the check if `notificationManager`, you can remove it, in a case where `notificationManager` is `null` we should at least add a `log` or let the code fail we will see it more quickly then trying understanding why we don't receive the notification mail

Reply:
I thought I had an NPE at some point but I didn't when replayed now so I removed it.